fuzzers/057-pip-bi/generate.tcl

```diff
@@ -41,22 +41,50 @@ for {set idx 0} {$idx < [llength $todo_lines]} {incr idx} {
     set dst_wire [lindex $line 1]
     set src_wire [lindex $line 2]
 
-    if {$tile_type == "INT_L"} {set tile [lindex $int_l_tiles $idx]; set other_tile [lindex $int_r_tiles $idx]}
-    if {$tile_type == "INT_R"} {set tile [lindex $int_r_tiles $idx]; set other_tile [lindex $int_l_tiles $idx]}
-
-    puts "PIP Tile: $tile"
-
-    set driver_site [get_sites -of_objects [get_site_pins -of_objects [get_nodes -downhill \
-            -of_objects [get_nodes -of_objects [get_wires $other_tile/CLK*0]]]]]
-
-    puts "LUT Tile (Site): $other_tile ($driver_site)"
-
-    set mylut [create_cell -reference LUT1 mylut_$idx]
-    set_property -dict "LOC $driver_site BEL A6LUT" $mylut
-
-    set mynet [create_net mynet_$idx]
-    connect_net -net $mynet -objects "$mylut/I0 $mylut/O"
-    route_via $mynet "$tile/$src_wire $tile/$dst_wire"
+    set tries 0
+    while {1} {
+        set tile_idx [expr $tries + [expr $idx * 3]]
+        incr tries
+
+        if {$tile_type == "INT_L"} {set tile [lindex $int_l_tiles $tile_idx]; set other_tile [lindex $int_r_tiles $idx]}
+        if {$tile_type == "INT_R"} {set tile [lindex $int_r_tiles $tile_idx]; set other_tile [lindex $int_l_tiles $idx]}
+
+        puts "PIP Tile: $tile"
+
+        set driver_site [get_sites -of_objects [get_site_pins -of_objects [get_nodes -downhill \
+                -of_objects [get_nodes -of_objects [get_wires $other_tile/CLK*0]]]]]
+
+        puts "LUT Tile (Site): $other_tile ($driver_site)"
+
+        set mylut [create_cell -reference LUT1 mylut_$idx]
+        set_property -dict "LOC $driver_site BEL A6LUT" $mylut
+
+        set mynet [create_net mynet_$idx]
```

Comment:
`mynet_$idx` net should be created once before the loop, otherwise this will fail on retry

Reply:
Good catch, hoisted out of loop.